File: Pessoais/numeros_primos.rb


class Primo
  attr_accessor :number

    def initialize(number)
      self.number = number
    end


      def Result

      dividendo = 1
      result = 0

        return puts "Digite um número maior que 0" if self.number === 0

        while (dividendo <= self.number)

          if(self.number % dividendo === 0)
            result+=1

          end
            dividendo+=1

        end

    puts  result === 2 ? "#{self.number} é um número Primo" : "#{self.number} não é um número Primo"

  end

end

Primo.new(4).Result()

File: Pessoais/numeros_primos.js


class Primo{

constructor(number){

this.number = number

}
    Result(){
        let dividendo = 1
        let result = 0

        if(this.number === 0)return console.log("Digite um valor maior que 0")

            while(dividendo <= this.number){
                if(this.number % dividendo === 0){
                   result++
                }
                dividendo++
            }
            return console.log(result === 2 ? `${this.number} é um número Primo` : `${this.number} não é um número Primo`)
    }


}

new Primo(4).Result()

